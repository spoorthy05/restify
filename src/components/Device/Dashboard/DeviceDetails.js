import React from 'react';
import DeviceDetail from './DeviceDetail';

const deviceDetails  = (props) => {
    let filterText = props.filterText;
    var devices = [];
    let deviceData = props.data.data;
    if(deviceData) {
        devices = deviceData.map(function(device) {
            if (device.name.indexOf(filterText) === -1) {
              return;
            }
            return (<DeviceDetail onStatusChange={(event, device) => props.onStatusChange(event, device)}
                     key={device.name} device={device}/>);
        });
    } 
    
    return(
        <table className="DeviceTable">
            <thead>
                <tr>
                <th>NAME</th>
                <th>UNIT</th>
                <th>VALUE</th>
                <th>TIMESTAMP</th>
                <th>STATUS</th>
                </tr>
            </thead>
            <tbody>
               {devices}
            </tbody>
        </table>
    )
}

export default deviceDetails;