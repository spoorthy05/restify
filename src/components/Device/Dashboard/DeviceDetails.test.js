import React from 'react';
import DeviceDetails from './DeviceDetails';
import { shallow } from 'enzyme';

describe('DeviceDetails', () => {
    let data = {
        "data": [
            {
                "name": "acceleration_x",
                "unit": "m/s2",
                "value": 25.993848858558,
                "timestamp": 1556550441396,
                "active": true
            },
            {
                "name": "acceleration_y",
                "unit": "m/s2",
                "value": -128.993848858558,
                "timestamp": 1556550446366,
                "active": false
            }]
    }

    let noData = {"data" : []};
    it('should render correctly with no props', () => {
        const component = shallow(<DeviceDetails {...data}/>);
        component.setProps(data);
        expect(component).toMatchSnapshot();
      });

    it('should render correctly with empty props', () => {
        const component = shallow(<DeviceDetails {...noData}/>);
        component.setProps(noData);
        expect(component).toMatchSnapshot();
    });
});
