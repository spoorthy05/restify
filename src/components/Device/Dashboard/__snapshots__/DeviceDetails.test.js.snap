// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceDetails should render correctly with empty props 1`] = `ShallowWrapper {}`;

exports[`DeviceDetails should render correctly with no props 1`] = `ShallowWrapper {}`;
